applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_interface_element.cpp

```diff
@@ -64,62 +64,32 @@ int UPwSmallStrainInterfaceElement<TDim, TNumNodes>::Check(const ProcessInfo& rC
 
-    const CheckProperties check_properties(r_properties, "property at element", this->Id(),
+    const PropertiesType& r_properties = this->GetProperties();
+    const CheckProperties check_properties(r_properties, "property at element", element_Id,
                                            CheckProperties::Bounds::AllInclusive);
     check_properties.SingleUseBounds(CheckProperties::Bounds::AllExclusive).Check(MINIMUM_JOINT_WIDTH);
-
+    check_properties.CheckAvailability(IGNORE_UNDRAINED);
     if (!r_properties[IGNORE_UNDRAINED]) {
         check_properties.Check(TRANSVERSAL_PERMEABILITY);
         check_properties.SingleUseBounds(CheckProperties::Bounds::AllExclusive).Check(BULK_MODULUS_FLUID);
         check_properties.SingleUseBounds(CheckProperties::Bounds::AllExclusive).Check(DYNAMIC_VISCOSITY);
     }
 
-    // Verify the constitutive law
-    KRATOS_ERROR_IF_NOT(r_properties.Has(CONSTITUTIVE_LAW))
-        << "CONSTITUTIVE_LAW has Key zero or is not defined at element " << this->Id() << std::endl;
-
-    if (r_properties[CONSTITUTIVE_LAW]) {
-        // Verify compatibility of the element with the constitutive law
-        ConstitutiveLaw::Features LawFeatures;
-        r_properties[CONSTITUTIVE_LAW]->GetLawFeatures(LawFeatures);
-        bool correct_strain_measure = false;
-        for (unsigned int i = 0; i < LawFeatures.mStrainMeasures.size(); ++i) {
-            if (LawFeatures.mStrainMeasures[i] == ConstitutiveLaw::StrainMeasure_Infinitesimal)
-                correct_strain_measure = true;
-        }
-        KRATOS_ERROR_IF_NOT(correct_strain_measure)
-            << "constitutive law is not compatible with the element type "
-               "StrainMeasure_Infinitesimal "
-            << this->Id() << std::endl;
-
-        // Check constitutive law
-        ierr = r_properties[CONSTITUTIVE_LAW]->Check(r_properties, this->GetGeometry(), rCurrentProcessInfo);
-    } else
-        KRATOS_ERROR << "A constitutive law needs to be specified for the "
-                        "element "
-                     << this->Id() << std::endl;
-
-    const SizeType strain_size = this->GetProperties().GetValue(CONSTITUTIVE_LAW)->GetStrainSize();
-    if (TDim == 2) {
-        KRATOS_ERROR_IF_NOT(strain_size == 2)
-            << "Wrong constitutive law used. This is a 2D element! expected "
-               "strain size is 2 (el id = ) "
-            << this->Id() << std::endl;
-    } else {
-        KRATOS_ERROR_IF_NOT(strain_size == 3)
-            << "Wrong constitutive law used. This is a 3D element! expected "
-               "strain size is 3 (el id = ) "
-            << this->Id() << std::endl;
-    }
+    check_properties.CheckAvailabilityAndSpecified(CONSTITUTIVE_LAW);
+    ierr = r_properties[CONSTITUTIVE_LAW]->Check(r_properties, this->GetGeometry(), rCurrentProcessInfo);
+
+    ConstitutiveLawUtilities::CheckHasStrainMeasure_Infinitesimal(r_properties, element_Id);
+
+    const auto expected_size = VOIGT_SIZE_2D_INTERFACE;
```

Comment:
It's indeed a shame for this legacy interface element we don't have the correct StressStatePolicy. Since it's an element that we expect will be removed at some point, it's not worth the effort to create the correct policy for it. But because we don't know yet when exactly when this will be removed in favor of the surface interface elements we created, I would keep the functionality exactly as is for this one (although the code isn't pretty and a bit repetitive), such that it would still work in both 2D and 3D: 
```suggestion
    constexpr auto expected_size = TDim == 2 ? VOIGT_SIZE_2D_INTERFACE : VOIGT_SIZE_3D_INTERFACE;
```

Reply:
Thank you. changed